diff --git a/setup.py b/setup.py
index c96aece2..c969bad0 100644
--- a/setup.py
+++ b/setup.py
@@ -63,6 +63,8 @@ def _supported_unix():
                 libraries.append('unwind-x86')
         elif platform.machine() == 'aarch64':
             libraries.append('unwind-aarch64')
+        elif platform.machine() == 'ppc64le':
+            libraries.append('unwind-ppc64')
         else:
             raise NotImplementedError("unknown platform.machine(): %s" %
                                       platform.machine())
diff --git a/src/_vmprof.c b/src/_vmprof.c
index 5a8c4519..c2d2977e 100644
--- a/src/_vmprof.c
+++ b/src/_vmprof.c
@@ -40,6 +40,8 @@ PY_EVAL_RETURN_T * vmprof_eval(PY_STACK_FRAME_T *f, int throwflag)
     register PY_STACK_FRAME_T * callee_saved asm("r4");
 #elif defined(__aarch64__)
     register PY_STACK_FRAME_T * callee_saved asm("x19");
+#elif defined(__powerpc64__)
+    register PY_STACK_FRAME_T * callee_saved asm("r3");
 #else
 #    error "platform not supported"
 #endif
@@ -51,6 +53,8 @@ PY_EVAL_RETURN_T * vmprof_eval(PY_STACK_FRAME_T *f, int throwflag)
         "mov %1, %0\t\n"
 #elif defined(__arm__) || defined(__aarch64__)
 	"mov %1, %0\t\n"
+#elif defined(__powerpc64__)
+	"addi %1, %0, 0\t\n"
 #else
 #    error "platform not supported"
 #endif
diff --git a/src/vmprof_config.h b/src/vmprof_config.h
index 69a1b585..35d62512 100644
--- a/src/vmprof_config.h
+++ b/src/vmprof_config.h
@@ -26,6 +26,8 @@
   #define PC_FROM_UCONTEXT uc_mcontext.psw.addr
 #elif defined(__aarch64__)
   #define PC_FROM_UCONTEXT uc_mcontext.pc
+#elif defined(__powerpc64__)
+  #define PC_FROM_UCONTEXT uc_mcontext.gp_regs[PT_NIP]
 #else
   /* linux, gnuc */
   #define PC_FROM_UCONTEXT uc_mcontext.gregs[REG_RIP]
