From b064b73dc859e761a54d3e40711d5f06f5ef1d71 Mon Sep 17 00:00:00 2001
From: yanmeng <jnyanmeng@163.com>
Date: Fri, 31 Dec 2021 17:00:14 +0800
Subject: [PATCH] support apple M1 CPU

---
 src/vmprof_config.h | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/vmprof_config.h b/src/vmprof_config.h
index 35d62512..f1744cf3 100644
--- a/src/vmprof_config.h
+++ b/src/vmprof_config.h
@@ -13,11 +13,15 @@
 #elif defined(__OpenBSD__)
 #define PC_FROM_UCONTEXT sc_rip
 #elif defined( __APPLE__)
-  #if ((ULONG_MAX) == (UINT_MAX))
-    #define PC_FROM_UCONTEXT uc_mcontext->__ss.__eip
-  #else
-    #define PC_FROM_UCONTEXT uc_mcontext->__ss.__rip
-  #endif
+    #if ((ULONG_MAX) == (UINT_MAX))
+        #define PC_FROM_UCONTEXT uc_mcontext->__ss.__eip
+    #else
+        #if defined(__arm__) || defined(__arm64__)
+            #define PC_FROM_UCONTEXT uc_mcontext->__ss.__pc
+        #else
+            #define PC_FROM_UCONTEXT uc_mcontext->__ss.__rip
+        #endif
+    #endif
 #elif defined(__arm__)
   #define PC_FROM_UCONTEXT uc_mcontext.arm_ip
 #elif defined(__linux) && defined(__i386) && defined(__GNUC__)
