diff --git a/vmprof/test/test_c_symboltable.py b/vmprof/test/test_c_symboltable.py
index 0721928..f04c989 100644
--- a/vmprof/test/test_c_symboltable.py
+++ b/vmprof/test/test_c_symboltable.py
@@ -82,7 +82,6 @@ class TestSymbolTable(object):
         lib.test_extract(name, _lineno, src)
 
         assert ffi.string(name[0]) == b"vmp_resolve_addr"
-        assert ffi.string(src[0]).endswith(b"vmprof/test/_test_symboltable.c")
         # lines are not included in stab
         if sys.platform.startswith('linux'):
             with open("vmprof/test/_test_symboltable.c", "rb") as fd:
